distributed/client.py

```diff
@@ -1611,6 +1612,35 @@ def map(
                 "Dask no longer supports mapping over Iterators or Queues."
                 "Consider using a normal for loop and Client.submit"
             )
+        total_length = sum(len(x) for x in iterables)
+
+        if batch_size and batch_size > 1 and total_length > batch_size:
+            return sum(
+                [
+                    self.map(
+                        func,
+                        batch,
+                        key=key,
+                        workers=workers,
+                        retries=retries,
+                        priority=priority,
+                        allow_other_workers=allow_other_workers,
+                        fifo_timeout=fifo_timeout,
+                        resources=resources,
+                        actor=actor,
+                        actors=actors,
+                        pure=pure,
+                        **kwargs,
+                    )
+                    for batch in partition_all(batch_size, *iterables)
```

Comment:
It looks like `toolz.partition_all` doesn't accept more than one sequence argument, so this will fail for something like ...

```python
client.map(operator.add, range(100), range(100), batch_size=10)
```

(sorry for not finding this earlier)

However, what's here is still an improvement I think.  I'd be happy merging it in, especially given the long review cycle, if desired.

Reply:
Good catch. Hopefully resolved now.